MekHQ/src/mekhq/gui/PersonnelTab.java

```diff
@@ -51,377 +51,377 @@
 
-    private void changePersonnelView() {
-        final PersonnelTabView view = (choicePersonView.getSelectedItem() == null)
-                ? PersonnelTabView.GENERAL : choicePersonView.getSelectedItem();
-        final XTableColumnModel columnModel = (XTableColumnModel) getPersonnelTable().getColumnModel();
-        getPersonnelTable().setRowHeight(15);
-
-        // set the renderer
-        for (final PersonnelTableModelColumn column : PersonnelTableModel.PERSONNEL_COLUMNS) {
-            final TableColumn tableColumn = columnModel.getColumnByModelIndex(column.ordinal());
-            tableColumn.setCellRenderer(getPersonModel().getRenderer(choicePersonView.getSelectedItem()));
-            tableColumn.setPreferredWidth(column.getWidth());
-            columnModel.setColumnVisible(tableColumn,
-                    column.isVisible(getCampaign(), view, getPersonnelTable()));
-        }
+      personnelTable.setName("personnelTable");
+      preferences.manage(new JTablePreference(personnelTable));
+    } catch (Exception ex) {
+      LogManager.getLogger().error("Failed to set user preferences", ex);
     }
-
-    public void focusOnPerson(UUID id) {
-        splitPersonnel.resetToPreferredSizes();
-        int row = -1;
-        for (int i = 0; i < personnelTable.getRowCount(); i++) {
-            if (personModel.getPerson(personnelTable.convertRowIndexToModel(i)).getId().equals(id)) {
-                row = i;
-                break;
-            }
-        }
-        if (row == -1) {
-            // try expanding the filter to all units
-            choicePerson.setSelectedIndex(0);
-            for (int i = 0; i < personnelTable.getRowCount(); i++) {
-                if (personModel.getPerson(personnelTable.convertRowIndexToModel(i)).getId().equals(id)) {
-                    row = i;
-                    break;
-                }
-            }
-
-        }
-        if (row != -1) {
-            personnelTable.setRowSelectionInterval(row, row);
-            personnelTable.scrollRectToVisible(personnelTable.getCellRect(row, 0, true));
-        }
+  }
+
+  /* For export */
+  public JTable getPersonnelTable() {
+    return personnelTable;
+  }
+
+  public PersonnelTableModel getPersonModel() {
+    return personModel;
+  }
+
+  /*
+   * (non-Javadoc)
+   *
+   * @see mekhq.gui.CampaignGuiTab#refreshAll()
+   */
+  @Override
+  public void refreshAll() {
+    refreshPersonnelList();
+    changePersonnelView();
+  }
+
+  public void filterPersonnel() {
+    final PersonnelFilter filter = (choicePerson.getSelectedItem() == null)
+        ? PersonnelFilter.ACTIVE : choicePerson.getSelectedItem();
+    personnelSorter.setRowFilter(new RowFilter<>() {
+      @Override
+      public boolean include(Entry<? extends PersonnelTableModel, ? extends Integer> entry) {
+        return filter.getFilteredInformation(entry.getModel().getPerson(entry.getIdentifier()),getCampaignGui().getCampaign().getLocalDate());
```

Comment:
Style change request: spaces between parameters in parameter lists for readability.